### PR TITLE
Add package.json and (and tsconfig.json) for TS/JS language config roots

### PR DESCRIPTION
**Pull request by jpaju**

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -703,6 +703,7 @@ scope = "source.jsx"
 injection-regex = "jsx"
 language-id = "javascriptreact"
 file-types = ["jsx"]
+roots = [ "package.json" ]
```
Comment (the-mikedavis) [inline languages.toml:706]: Should this be added for JavaScript as well?

Comment (jpaju) [reply languages.toml:707]: Ohh yeah, that was definitely my intention but somehow that wasn't inlcuded in the commit. Should be fixed now!

**Review by the-mikedavis — APPROVED** [review]

**Comment by archseer** [comment]
Hmm won't this break package workspaces though? Usually you want the top level package.json/git root there

**Comment by archseer** [comment]
See https://github.com/helix-editor/helix/pull/5748

**Comment by the-mikedavis** [comment]
I believe we use the topmost directory with a root marker: https://github.com/helix-editor/helix/blob/beb5afcbef9f102c209de7aa32f126ded9dda515/helix-lsp/src/lib.rs#L1009-L1066

so if we a workspace like

```
.
+-- package.json
`-- packages
   +-- a
   |   `-- package.json
   `-- b
       `-- package.json
```

we should accept that top-most `package.json`. I believe we have similar roots set up for Elixir for example with `mix.exs` or `mix.lock` that should still work inside umbrella workspaces

**Comment by jpaju** [comment]
I tried this out and can confirm that the behavior is what @the-mikedavis described! When `package.json` is specified as `roots` in `languages.toml`, Helix reports the correct directory in `rootUri`/`rootPath`/`workspaceFolders`. This applies both to the monorepo folder structure I initially described and to workspace folder structure that @archseer described that has a top-level `package.json`.

**Pull request merged**